core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -969,6 +972,26 @@ protected Object visitNotExpression(NotExpression node, Object context)
         @Override
         protected Object visitLogicalExpression(LogicalExpression node, Object context)
         {
+            // Try simplifying expressions using TupleDomain, e.g. a > 1 and a > 2 => a > 2
+            PushPredicateIntoTableScan.SplitExpression splitExpression = splitExpression(plannerContext, node);
+            if (splitExpression.getNonDeterministicPredicate() == TRUE_LITERAL) {
+                DomainTranslator.ExtractionResult decomposedPredicate = DomainTranslator.getExtractionResult(
+                        plannerContext,
+                        session,
+                        splitExpression.getDeterministicPredicate(),
+                        typeProvider);
+                DomainTranslator domainTranslator = new DomainTranslator(plannerContext);
+                if (decomposedPredicate.getRemainingExpression() == TRUE_LITERAL) {
```

Comment:
we could combine remaining expression with the one obtained from TupleDomain
thus we could simply the part of expression that can be simplified, while retaining all the rest intact.